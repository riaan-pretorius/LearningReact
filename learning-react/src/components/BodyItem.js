import React, { Component } from 'react';
import { Col } from 'reactstrap';


class BodyItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            heading: props.heading,
            subText: props.subText
        }
    }
    render() {
        return (
            <Col sm="6">
                <h4>{this.state.heading}</h4>
                <p>{this.state.subText}</p>
            </Col>
        )
    }
}

export default BodyItem