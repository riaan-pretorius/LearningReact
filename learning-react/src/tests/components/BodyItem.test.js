import React from 'react'
import BodyItem from '../../components/BodyItem'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'

test('BodyItem Test Snapshot', () => {
    const component = renderer.create(
    <BodyItem heading="Testing" subText="Testing SubText" />
    )
    let test = component.toJSON();
    expect(test).toMatchSnapshot();
    //expect(test.props.heading).toBe("Testing");
    //expect(test.props.subText).toBe("Testing SubText");
})

test('BodyItem Test', () => {
    const component = mount(
        <BodyItem heading="Testing" subText="Testing SubText" />
    )
    expect(component.find('.col').exists()).toBeTruthy()

    let header = component.find('h4')
    expect(header.exists()).toBeTruthy()
    expect(header.text()).toBe("Testing")

    let p = component.find('p')
    expect(p.exists()).toBeTruthy()
    expect(p.text()).toBe("Testing SubText")
})