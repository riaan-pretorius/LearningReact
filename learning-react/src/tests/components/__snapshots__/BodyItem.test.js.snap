// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyItem Test Snapshot 1`] = `
<div
  className="col col-sm-6"
>
  <h4>
    Testing
  </h4>
  <p>
    Testing SubText
  </p>
</div>
`;
